components/LogWorkout/LogWorkout.jsx

```diff
@@ -77,6 +81,56 @@ const LogWorkout = (props) => {
 
   const [exerciseState, setExerciseState] = useState(initialExerciseState);
 
+  // const currentUser = firebase.auth().currentUser.uid;
+  const currentUser = '68w6wWz8l5QJO3tDukh1fRXWYjD2';
+
+  const dbRef = firebase.firestore();
+  const userRef = dbRef.collection('users').doc(currentUser);
+
+  const incrementSelectedCycleIdx = () => {
+    // userRef.update({selectedCycleIndex: 2})
+    // userRef.get().then(function(doc) {
+    //       console.log("Document data:", doc.data().selectedCycleIndex);
+    //   } 
+    // )
+    return (
+    <CurrentCycle
+          rightPress={() => { dispatch({ type: INCREMENT_SELECTED_CYCLE_INDEX, cycleLength: cycleDetails.length }); }}
+        /> )
+  };
```

Comment:
I don't think this does what you want it to do. You just need a function that calls dispatch to increment selected cycle index and also updates the database as well. The stuff ou have commented out on line 91 is pretty close for the database part